### ISSUE TITLE
False boolean does not override true boolean when merging (2.4.1)

### ISSUE THREAD

**Issue by FranckKe** [mikefarah/yq#512]
Thank you for developing `yq` :)

**Description**
Merging a `false` boolean into a `true` boolean does not work as expected.

version of yq: `2.4.1`
operating system: Docker (`mikefarah/yq:2`)

**Input Yaml**
data1.yml:
```yaml
key:
  value: false
```

data2.yml:
```yaml
key:
  value: true
```

**Command**
The command you ran:
```
yq merge data1.yml data2.yml
```

**Actual behavior**

```yaml
key1:
  value: true

```

**Expected behavior**

```yaml
key:
  value: false
```

**Additional context**
It works fine with yq 3+. I'm using 2 until https://github.com/mikefarah/yq/issues/455 is addressed.
My workaround is to cast boolean as a string and remove quotes after the merging is done.

**Comment by mikefarah** [comment]
#455 has been fixed in 3.4.0

**Comment by mikefarah** [comment]
Marking as 'wontfix' as I'm no longer supporting v2
